packages/sdk/explorer/src/block-explorer.ts

```diff
@@ -114,23 +123,28 @@ export class BlockExplorer {
     }
   }
 
-  async tryParseTxInput(address: string, input: string): Promise<null | CallDetails> {
-    const callSignature = input.slice(0, 10)
-    const { contract: contractName, abi: matchedAbi } = this.getContractMethodAbi(
-      address,
-      callSignature
-    )
-    if (!contractName || !matchedAbi) {
-      return null
+  getKnownFunction(selector: string): ABIDefinition | undefined {
+    // TODO(bogdan): This could be replaced with a call to 4byte.directory
+    // or a local database of common functions.
+    const knownFunctions: { [k: string]: string } = {
```

Comment:
Mmm, we could eventually have a core contract that maps selector to signature. If I'm understanding correctly, this will be generated before executing it, so "filling" that other core contract, should be a different proposal that has to be executed before the one that actually uses it. But could be a nice to have for the future. But that way we can rely on our own blockchain, and things that already passed a governance check

Reply:
Not exactly sure if this is what you mean, but we could have a core contract that acts as a canonical source of info for external contract calls from governance, with something like: 
```
mapping(address => ExternalContractInfo) externalContracts;

struct ExternalContractInfo {
  string name;
  mapping(bytes4 => ExternalFunctionInfo) functions;
}

struct ExternalFunctionInfo {
  string signature;
  bool allowed;
}
```

Which could be used to parse external calls, but maybe even act as a whitelist?

Reply:
Yes, exactly that. But as I said, this could be something to be added in the future and not in this PR